extensions/sockets/src/main/java/org/mule/module/socket/api/socket/SocketProperties.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.socket.api.socket;
+
+/**
+ * Interface for objects that provide common configuration for generic sockets.
+ *
+ * @since 4.0
+ */
+public interface SocketProperties
+{
+
+    /**
+     * The size of the buffer (in bytes) used when sending data, set on the socket itself.
+     */
+    Integer getSendBufferSize();
```

Comment:
The original transport code says:

```java
    private Boolean sendTcpNoDelay = DEFAULT_SEND_TCP_NO_DELAY;
    private Boolean reuseAddress = DEFAULT_REUSE_ADDRESS;

    // These options are undefined by default so that java.net.Socket/java.net.ServerSocket defaults are used.  We do
    // however document the java.net.Socket/java.net.ServerSocket defaults in the schema for usability.
    private Integer serverTimeout ;
    private Integer timeout;
    private Integer linger;

    // These options are undefined by default so that by default it is the OS TCP/IP stack that configures, or dynamically
    // manages, these values.
    private Integer sendBufferSize;
    private Integer receiveBufferSize;
    private Integer receiveBacklog;
    private Boolean keepAlive;
```
 The use of boxed types allows null values, which tell us to use the socket's  default values.